### ISSUE TITLE
Brave does not automatically resume downloads

### ISSUE THREAD

**Issue by jdknox** [brave/brave-browser#23912]
I searched for "download resume" and "download network error" in the existing issues, and this bug didn't show up in the list on GitHub.

## Description 
Brave does not automatically resume downloads, even though the Brave Help Center explicitly states "Brave will automatically resume the download" on the [How do I fix file download errors?](https://support.brave.com/hc/en-us/articles/360018192491-How-do-I-fix-file-download-errors-) page.

## Steps to Reproduce

This issue can only be reproduced if there is a `Network Error`.  I don't know how to reproduce the necessary error.

## Actual result:
Brave never resumes the download unless the download is manually resumed (and the download _does_ indeed resume when manually clicking the `Resume` button).

## Expected result:
Brave should automatically resume the download as soon as the network is functioning properly.

## Reproduces how often: 
No steps to reproduce, but it happens to me every time the download fails.

## Brave version (brave://version info)
[Version 1.40.113 Chromium: 103.0.5060.114 (Official Build) (64-bit)](https://brave.com/latest/)

**Comment by jdknox** [comment]
If clicking the `Resume` button works, does that mean the server allows automatic resuming in addition to manual resume?  Put another way: does Brave's _automatic_ resuming require explicit server support?  If so, I would say the documentation should be changed to reflect the requirement.

**Comment by gruber123** [comment]
No, you are wrong the documentation is right described for download resuming, and yes, you are right download resuming must be explicitly affected on the server, for simple you can resume only simple links rhat access dowload file and still if server support resuming, that feature works in most browsers

**Comment by jdknox** [comment]
But my question is about automatic resuming.  I know the server I was connected to supports resuming because I was able to resume.  But Brave was not automatically resuming as stated in the docs.

Maybe I should take a look at the source code to see when the automatic resuming is supposed to happen.  Would you happen to know where I should start looking?

**Comment by jdknox** [comment]
Yes, I mean to resume after connection is lost.  Brave shows "Failed - Network Error" on the `Downloads` page.  I will test to see if it resumes on Brave restart.

Doc link is in the initial post, but here also: https://support.brave.com/hc/en-us/articles/360018192491-How-do-I-fix-file-download-errors-

Here is the quote:

    1.     Make sure your internet connection is working normally.
    -     Brave will automatically resume the download. You can also resume the file download by clicking the Down arrow Down Arrow —>  Resume or Menu icon_MM.png —> Resume.

**Comment by jdknox** [comment]
I'm trying to confirm if the limit is actually being hit, which is preventing further auto-resume.  But the limit is not being hit, unless Brave has patched the Chromium histogram code that records the limit.
Chromium code that records the limit being reached is here: https://source.chromium.org/chromium/chromium/src/+/main:components/download/internal/common/download_stats.cc;drc=cfd951a304bb7a1e6424c614c094f5bc581ca8b7;l=256

But `Download.Resume.AutoResumeLimitReached.LastReason` is not showing up in my histogram log:

![brave_download_histograms](https://user-images.githubusercontent.com/926182/177830991-77c30221-4ab0-48b1-a1da-9a0c165470cd.png)

Also on that log, you can see the `Download.Resume.LastReason` has only one sample, and that was recorded exactly when I manually resumed.  The auto-resume feature is never triggered even once in Brave, or at least it is never recorded.

**Comment by leekaiwei** [comment]
Any update on this?
